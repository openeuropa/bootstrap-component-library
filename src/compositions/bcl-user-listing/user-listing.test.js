import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-user-listing/dataUserListing";

const template = "@oe-bcl/bcl-user-listing/user-listing.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - User listing", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });
});
