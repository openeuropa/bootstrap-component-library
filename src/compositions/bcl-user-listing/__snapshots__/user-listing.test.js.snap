// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - User listing renders correctly 1`] = `
<jest>
  <main>
    <div
      class="container mt-5"
    >
      <div
        class="row"
      >
        <div
          class="col-md-3"
        >
          <div
            aria-labelledby="offcanvasExampleLabel"
            class="bcl-offcanvas offcanvas-start offcanvas"
            data-bs-scroll="true"
            id="bcl-offcanvas"
            tabindex="-1"
          >
            <div
              class="offcanvas-header p-lg-0"
            >
              <h4
                class="offcanvas-title mb-lg-4"
                id="offcanvasExampleLabel"
              >
                Filter options
              </h4>
              <button
                class="btn-close text-reset d-lg-none"
                data-bs-dismiss="offcanvas"
                type="button"
              />
            </div>
            <div
              class="offcanvas-body p-lg-0"
            >
              <form
                novalidate="true"
                onsubmit="return false;"
              >
                <div
                  class="mb-3 pb-3 border-bottom"
                >
                  <input
                    class="form-control"
                    placeholder="Search by name or surname"
                    type="text"
                  />
                </div>
                <div
                  class="mb-3"
                >
                  <label
                    class="form-label"
                  >
                    Organisation
                  </label>
                  <select
                    class="form-select"
                  >
                    <option
                      value="1"
                    >
                      a select option
                    </option>
                    <option
                      value="2"
                    >
                      another select option
                    </option>
                    <option
                      value="3"
                    >
                      another option
                    </option>
                    <option
                      value="4"
                    >
                      last option
                    </option>
                  </select>
                </div>
                <div
                  class="mb-3"
                >
                  <label
                    class="form-label"
                  >
                    Profile Type
                  </label>
                  <select
                    class="form-select"
                  >
                    <option
                      value="1"
                    >
                      a select option
                    </option>
                    <option
                      value="2"
                    >
                      another select option
                    </option>
                    <option
                      value="3"
                    >
                      another option
                    </option>
                    <option
                      value="4"
                    >
                      last option
                    </option>
                  </select>
                </div>
                <div
                  class="mb-3 pb-3 border-bottom"
                >
                  <label
                    class="form-label"
                  >
                    Group
                  </label>
                  <select
                    class="form-select"
                  >
                    <option
                      value="1"
                    >
                      a select option
                    </option>
                    <option
                      value="2"
                    >
                      another select option
                    </option>
                    <option
                      value="3"
                    >
                      another option
                    </option>
                    <option
                      value="4"
                    >
                      last option
                    </option>
                  </select>
                </div>
                <div
                  class="mb-3"
                >
                  <label
                    class="form-label"
                  >
                    Option 1
                  </label>
                  <input
                    class="form-control"
                    placeholder="Placeholder"
                    type="text"
                  />
                </div>
                <div
                  class="mb-3"
                >
                  <label
                    class="form-label"
                  >
                    Option 2
                  </label>
                  <input
                    class="form-control"
                    placeholder="Placeholder"
                    type="text"
                  />
                </div>
                <div
                  class="mt-4"
                >
                  <button
                    class="me-2 btn btn-primary btn-md"
                    type="submit"
                  >
                    Refine
                  </button>
                  <button
                    class="btn btn-secondary btn-md"
                    type="submit"
                  >
                    Clear
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
        <div
          class="col-12 col-lg-9"
        >
          <button
            aria-controls="bcl-offcanvas"
            class="d-lg-none w-100 mb-4 btn btn-primary btn-md"
            data-bs-target="#bcl-offcanvas"
            data-bs-toggle="offcanvas"
            type="button"
          >
            Filter options
            <svg
              class="ms-2 ms-2-5 bi icon--fluid"
            >
              <use
                xlink:href="#sliders"
              />
            </svg>
          </button>
          <div
            class="mb-3 clearfix"
          >
            <div
              class="float-end d-none d-md-block"
            >
              <label
                class="form-label"
              >
                Sort by
              </label>
              <select
                class="mb-4 form-select"
              >
                <option
                  value="1"
                >
                  a select option
                </option>
                <option
                  value="2"
                >
                  another select option
                </option>
                <option
                  value="3"
                >
                  another option
                </option>
                <option
                  value="4"
                >
                  last option
                </option>
                <option
                  selected=""
                  value=""
                >
                  A-Z
                </option>
              </select>
            </div>
            <h4
              class="mb-4"
            >
              Users (6)
            </h4>
            <span
              class="me-2 badge bg-primary"
            >
              Name
              <span
                aria-hidden="true"
                class="icon--close"
              >
                <svg
                  class="bi icon--s"
                >
                  <use
                    xlink:href="#x"
                  />
                </svg>
              </span>
            </span>
            <span
              class="me-2 badge bg-primary"
            >
              Organisation
              <span
                aria-hidden="true"
                class="icon--close"
              >
                <svg
                  class="bi icon--s"
                >
                  <use
                    xlink:href="#x"
                  />
                </svg>
              </span>
            </span>
          </div>
          <hr
            class="d-none d-md-block mb-4"
          />
          <div
            class="d-md-none"
          >
            <label
              class="form-label"
            >
              Sort by
            </label>
            <select
              class="mb-4 form-select"
            >
              <option
                value="1"
              >
                a select option
              </option>
              <option
                value="2"
              >
                another select option
              </option>
              <option
                value="3"
              >
                another option
              </option>
              <option
                value="4"
              >
                last option
              </option>
              <option
                selected=""
                value=""
              >
                A-Z
              </option>
            </select>
          </div>
          <div
            class="bcl-listing bcl-listing--default-1-col"
          >
            <div
              class="row row-cols-1 g-4"
            >
              <div
                class="col"
              >
                <div
                  class="listing-item border-bottom border-md-0 border-0 card"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-3 col-lg-2 rounded mw-listing-img align-self-center"
                    >
                      <img
                        alt="alt img"
                        class="card-img-top mb-3"
                        src="https://picsum.photos/160?random=1"
                      />
                    </div>
                    <div
                      class="col-md-9 col-lg-10 align-self-center"
                    >
                      <div
                        class="card-body p-0 pb-md-0 pb-3"
                      >
                        <div
                          class="mb-3"
                        >
                          <span
                            class="badge bg-primary"
                          >
                            Profile Type
                          </span>
                        </div>
                        <h5
                          class="card-title mb-3"
                        >
                          Stefan Mayer
                        </h5>
                        <span
                          class="text-muted me-2"
                        >
                          Organisation
                        </span>
                        <span
                          class="text-muted me-2"
                        >
                          Position
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="col"
              >
                <div
                  class="listing-item border-bottom border-md-0 border-0 card"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-3 col-lg-2 rounded mw-listing-img align-self- align-self-center"
                    >
                      <img
                        alt="alt img"
                        class="card-img-top mb-3"
                        src="https://picsum.photos/160?random=2"
                      />
                    </div>
                    <div
                      class="col-md-9 col-lg-10 align-self- align-self-center"
                    >
                      <div
                        class="card-body p-0 pb-md-0 pb-3"
                      >
                        <div
                          class="mb-3"
                        >
                          <span
                            class="badge bg-primary"
                          >
                            Profile Type
                          </span>
                        </div>
                        <h5
                          class="card-title mb-3"
                        >
                          Adjur Tichar
                        </h5>
                        <span
                          class="text-muted me-2"
                        >
                          Organisation
                        </span>
                        <span
                          class="text-muted me-2"
                        >
                          Position
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="col"
              >
                <div
                  class="listing-item border-bottom border-md-0 border-0 card"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-3 col-lg-2 rounded mw-listing-img align-self- align-self- align-self-center"
                    >
                      <img
                        alt="alt img"
                        class="card-img-top mb-3"
                        src="https://picsum.photos/160?random=3"
                      />
                    </div>
                    <div
                      class="col-md-9 col-lg-10 align-self- align-self- align-self-center"
                    >
                      <div
                        class="card-body p-0 pb-md-0 pb-3"
                      >
                        <div
                          class="mb-3"
                        >
                          <span
                            class="badge bg-primary"
                          >
                            Profile Type
                          </span>
                        </div>
                        <h5
                          class="card-title mb-3"
                        >
                          Szép Mara
                        </h5>
                        <span
                          class="text-muted me-2"
                        >
                          Organisation
                        </span>
                        <span
                          class="text-muted me-2"
                        >
                          Position
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="col"
              >
                <div
                  class="listing-item border-bottom border-md-0 border-0 card"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-3 col-lg-2 rounded mw-listing-img align-self- align-self- align-self- align-self-center"
                    >
                      <img
                        alt="alt img"
                        class="card-img-top mb-3"
                        src="https://picsum.photos/160?random=4"
                      />
                    </div>
                    <div
                      class="col-md-9 col-lg-10 align-self- align-self- align-self- align-self-center"
                    >
                      <div
                        class="card-body p-0 pb-md-0 pb-3"
                      >
                        <div
                          class="mb-3"
                        >
                          <span
                            class="badge bg-primary"
                          >
                            Profile Type
                          </span>
                        </div>
                        <h5
                          class="card-title mb-3"
                        >
                          Daniel Cardoso Rocha
                        </h5>
                        <span
                          class="text-muted me-2"
                        >
                          Organisation
                        </span>
                        <span
                          class="text-muted me-2"
                        >
                          Position
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="col"
              >
                <div
                  class="listing-item border-bottom border-md-0 border-0 card"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-3 col-lg-2 rounded mw-listing-img align-self- align-self- align-self- align-self- align-self-center"
                    >
                      <img
                        alt="alt img"
                        class="card-img-top mb-3"
                        src="https://picsum.photos/160?random=5"
                      />
                    </div>
                    <div
                      class="col-md-9 col-lg-10 align-self- align-self- align-self- align-self- align-self-center"
                    >
                      <div
                        class="card-body p-0 pb-md-0 pb-3"
                      >
                        <div
                          class="mb-3"
                        >
                          <span
                            class="badge bg-primary"
                          >
                            Profile Type
                          </span>
                        </div>
                        <h5
                          class="card-title mb-3"
                        >
                          Aurelia Panicucci
                        </h5>
                        <span
                          class="text-muted me-2"
                        >
                          Organisation
                        </span>
                        <span
                          class="text-muted me-2"
                        >
                          Position
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="col"
              >
                <div
                  class="listing-item border-bottom border-md-0 border-0 card"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-3 col-lg-2 rounded mw-listing-img align-self- align-self- align-self- align-self- align-self- align-self-center"
                    >
                      <img
                        alt="alt img"
                        class="card-img-top mb-3"
                        src="https://picsum.photos/160?random=6"
                      />
                    </div>
                    <div
                      class="col-md-9 col-lg-10 align-self- align-self- align-self- align-self- align-self- align-self-center"
                    >
                      <div
                        class="card-body p-0 pb-md-0 pb-3"
                      >
                        <div
                          class="mb-3"
                        >
                          <span
                            class="badge bg-primary"
                          >
                            Profile Type
                          </span>
                        </div>
                        <h5
                          class="card-title mb-3"
                        >
                          Felicyta Kowalska
                        </h5>
                        <span
                          class="text-muted me-2"
                        >
                          Organisation
                        </span>
                        <span
                          class="text-muted me-2"
                        >
                          Position
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <nav
            class="mb-5 mt-4"
          >
            <ul
              class="pagination justify-content-center"
            >
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href="#"
                >
                  Previous
                </a>
              </li>
              <li
                class="page-item active"
              >
                <a
                  aria-current="page"
                  class="page-link"
                  href="#"
                >
                  1
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href="#"
                >
                  2
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href="#"
                >
                  3
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href="#"
                >
                  4
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href="#"
                >
                  5
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href="#"
                >
                  Next
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </main>
</jest>
`;
